src/sentry/snuba/metrics/fields/base.py

```diff
@@ -185,37 +197,33 @@ def run_post_query_function(self, data, idx=None):
         raise NotImplementedError
 
 
-class RawAggregatedMetric(MetricFieldBase):
+@dataclass
+class MetricExpressionDefinition:
+    op: str
+    metric_name: str
+
+
+class MetricExpression(MetricExpressionDefinition, MetricExpressionBase, ABC):
     """
     This class serves the purpose of representing any aggregate, raw metric combination for
     example `sum(sentry.sessions.session)`. It is created on the fly to abstract the field
     conversions to SnQL away from the query builder.
     """
 
+    @classmethod
+    def from_dict(cls, op, metric_name):
```

Comment:
With `from_dict` I would expect a parameter of type `dict`, from which the instance is constructed, e.g. https://github.com/getsentry/snuba/blob/df261841594c7d25b87a4a4c80eb55402e0d2430/snuba/snapshots/__init__.py#L24. But it's just a nitpick, so up to you.

Reply:
Yeah agreed

Reply:
`from_` is a very popular convention, at the same time i agree about teh `dict` part, maybe also `parse` could work since we return different types?

type hints would be really good in this module in general